titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/KubeUtil.java

```diff
@@ -209,4 +213,24 @@ public static String getNodeIpV4Address(V1Node node) {
                 .map(V1NodeAddress::getAddress)
                 .orElse("UnknownIpAddress");
     }
+
+    public static Map<String, String> createPodAnnotations(
+            Job<?> job,
+            byte[] containerInfoData,
+            Map<String, String> passthroughAttributes,
+            boolean includeJobDescriptor
+    ) {
+        String encodedContainerInfo = Base64.getEncoder().encodeToString(containerInfoData);
+
+        Map<String, String> annotations = new HashMap<>(passthroughAttributes);
+        annotations.putAll(PerformanceToolUtil.toAnnotations(job));
+        annotations.put("containerInfo", encodedContainerInfo);
+
+        if (includeJobDescriptor) {
+            String jobDescriptorJson = CommonObjectMappers.writeValueAsString(ObjectMappers.storeMapper(), job.getJobDescriptor());
```

Comment:
minor: use `ObjectMappers.storeMapper()#writeValue(OutputStream, Object)` to save some heap space, avoid `String` allocations and unnecessary UTF-16 shenanigans.

That way you can also pass in a `GZIPOutputStream` directly to the `writeValue()` method and allow chunked encoding of the whole jobDescriptor.

Reply:
I like this idea, but will address it in a followup PR.